jetty-core/jetty-core-server/src/main/java/org/eclipse/jetty/core/server/handler/gzip/GzipHandler.java

```diff
@@ -72,32 +72,38 @@ public long getContentLength()
                     // TODO hide the content length
                     return -1;
                 }
-
-                @Override
-                public Content readContent()
-                {
-                    // TODO inflate data
-                    return super.readContent();
-                }
-
-                @Override
-                public Response accept()
-                {
-                    Response response = super.accept();
-                    if (response != null)
+            },
+            (r, processor) ->
+                acceptor.accept(r, exchange ->
+                    processor.process(new Exchange.Wrapper(exchange)
                     {
-                        return new Response.Wrapper(request, request.accept())
+                        @Override
+                        public Response getResponse()
                         {
-                            @Override
-                            public void write(boolean last, Callback callback, ByteBuffer... content)
+                            return new Response.Wrapper(exchange.getRequest(), super.getResponse())
```

Comment:
Wrapping the acceptor to wrap the processor to wrap the Exchange to wrap the response to intercept write calls!!!!